x-pack/plugins/lens/public/datatable_visualization/expression.test.tsx

```diff
@@ -106,7 +141,7 @@ describe('datatable_expression', () => {
           },
         ],
         negate: true,
-        timeFieldName: undefined,
+        timeFieldName: 'a',
```

Comment:
This doesn't look correct - `a` is a string field but it ends up as the "timeFieldName". 
Seems like the logic here https://github.com/elastic/kibana/pull/75184/files#diff-9579ca4b06d409542e1c027008518abe610abdd3bf7a3b897f408fec7a324e44R177 is flawed. Not blocking this PR as it doesn't seem like it got introduced here, but we should probably fix it.